### DISCUSSION TITLE
with-mux-video   Error : Objects are not valid as a React child

### DISCUSSION THREAD

**Discussion by EhanDuan** [vercel/next.js#37411] [Help] [answered]
Hi,

 I am deploying the `with-mux-video` example. What I have done:
+ create the example project:  
```shell
npx create-next-app --example with-mux-video with-mux-video-app
```
+ install dependencies with legacy since npm install error with cannot resolve ....: 
```shell
npm install  --legacy-peer-deps
```
+ Set up environment variables
> Copy the .env.local.example file in this directory to .env.local (which will be ignored by Git):
```shell
cp .env.local.example .env.local
```

Then I run the app, I select a file, it popped up error. Here is the screenshot. 
<img width="850" alt="image" src="https://user-images.githubusercontent.com/75397936/171698859-270b9a65-1432-4e8d-ba02-9a6464fae958.png">

Anyone have an idea about that?

**Comment by icyJoseph** [comment, marked as answer]
It'd seem that the error object has changed, detail in particular, you ought to do `setErrorMessage(err.detail.message)`, but I reckon that's only the first error to fix here.

Looking at the `upchunk` module, I see no changes, but yeah, detail is an object, and the `errorMessage` state must be a string, at least when passed to the `ErrorMessage` component.

#### 4 years ago

`detail` was a string, looking through the git blame on https://github.com/muxinc/upchunk/blob/master/src/upchunk.ts#L290

```js
    this.eventTarget.dispatchEvent(
      new CustomEvent('error', {
        detail: `An error occured uploading chunk ${
          this.chunkCount
        }. No more retries, stopping upload`,
      })
    );
```

#### Today

`detail` is an object

```js
this.dispatch('error', {
      message: `An error occured uploading chunk ${this.chunkCount}. No more retries, stopping upload`,
      chunk: this.chunkCount,
      attempts: this.attemptCount,
    }
```

#### Resolution

Read `err.detail.message`, and additionally, perhaps you want to make a PR to the examples repo, and update the code? 😄

**Reply by icyJoseph** [reply]
I made the PR now, hopefully it gets merged soon :) https://github.com/vercel/next.js/pull/37434
